Objects/typeobject.c

```diff
@@ -6156,6 +6161,16 @@ type_ready_inherit_as_structs(PyTypeObject *type, PyTypeObject *base)
     }
 }
 
+static void
+inherit_patma_flags(PyTypeObject *type, PyTypeObject *base) {
+    if (type->tp_flags & COLLECTION_FLAGS) {
+        return;
+    }
+    if ((base->tp_flags & COLLECTION_FLAGS) == 0) {
+        return;
+    }
+    base->tp_flags |= base->tp_flags & COLLECTION_FLAGS;
```

Comment:
I think you mean `type->tp_flags` on the LHS of the last assignment, right?

Also, the `base->tp_flags & COLLECTION_FLAGS` check seems unnecessary.

```suggestion
    type->tp_flags |= base->tp_flags & COLLECTION_FLAGS;
```